### ISSUE TITLE
Remove irrelevant fields from full manifest

### ISSUE THREAD

**Issue by mshadbolt** [DataBiosphere/azul#1368]
As a HCA metadata consumer I need to be able to access and understand the metadata tsv in order to consume and analyse HCA DCP data. Currently there are fields that appear in the metadata tsv that are not useful for a metadata consumer and make the tsv harder to use and understand. The largest tsv file has 206 columns ([HPSI Organoids](https://data.humancellatlas.org/explore/projects/005d611a-14d5-4fbf-846e-571a1f874f70)).

More consultation needs to be done to figure out which fields are useful but based on my experience as a metadata user I believe many of the fields below are not very useful for a data consumer, but happy for feedback to the contrary.
e.g.
```
analysis_file.provenance.schema_major_version
analysis_file.provenance.schema_minor_version
analysis_process.analysis_run_type
analysis_process.input_bundles
analysis_process.inputs.parameter_name
analysis_process.inputs.parameter_value
analysis_process.provenance.document_id
analysis_process.provenance.schema_major_version
analysis_process.provenance.schema_minor_version
analysis_process.reference_bundle
analysis_process.tasks.cpus
analysis_process.tasks.disk_size
analysis_process.tasks.docker_image
analysis_process.tasks.log_err
analysis_process.tasks.log_out
analysis_process.tasks.memory
analysis_process.tasks.start_time
analysis_process.tasks.stop_time
analysis_process.tasks.task_name
analysis_process.tasks.zone
analysis_process.timestamp_start_utc
analysis_process.timestamp_stop_utc
analysis_process.type.text
analysis_protocol.computational_method
analysis_protocol.provenance.document_id
analysis_protocol.provenance.schema_major_version
analysis_protocol.provenance.schema_minor_version
analysis_protocol.type.text
process.provenance.schema_major_version
process.provenance.schema_minor_version
project.funders.grant_id
project.funders.grant_title
project.funders.organization
```

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/AZUL-836)
┆Project Name: azul
┆Issue Number: AZUL-836

**Comment by hannes-ucsc** [comment]
> Currently there are fields that appear in the metadata tsv that are not useful for a metadata consumer and make the tsv harder to use and understand.

Let's say we remove these fields. How can we be sure that the next metadata user doesn't come forward and requests the reinclusion of these fields? In other words, how can we be sure that these fields aren't indeed. We should err on the side of including fields and only omit them if it is technically detrimental. 

I don't quite follow how including these fields makes the TSV harder to understand or use. The full manifest is not meant for human consumption, it's main purpose programmatic consumption and to be as complete as technically possible.

I would prefer to only remove a field if it is obvious that no one can do anything with it. The schema `*.provenance.schema_{major,minor}_version` field fall into that category. Most of the `analysis_process` fields don't in my opinion. The project fields, I'm on the fence.

**Comment by mshadbolt** [comment]
I partially agree it is for programmatic consumption but how does the person writing the program to interpret the data figure out what it means? The more fields there are the harder it is for a user to figure out what they actually need and write a program to use those fields. Also some users will not be programmatic, they will be a wet lab scientist with minimal programming experience who wants to see metadata. If the full manifest is not created for them then what metadata output are we creating that would meet their needs?

When would a downstream user need to know the number of CPUs or memory usage of an analysis that was carried out?

**Comment by hannes-ucsc** [comment]
> I partially agree it is for programmatic consumption but how does the person writing the program to interpret the data figure out what it means?

By reading the documentation. How does removing fields from the manifest make the manifest easier to understand? Easier to scan visually, yes, but there is no substitution for reading the documentation. Sometimes removal of columns make a manifest *harder* to understand because if the column was a discriminator between rows.

> Also some users will not be programmatic, they will be a wet lab scientist with minimal programming experience who wants to see metadata. If the full manifest is not created for them then what metadata output are we creating that would meet their needs?

One does not need programming experience to understand the full manifest. One needs to reading the documentation if something is not clear.

> When would a downstream user need to know the number of CPUs or memory usage of an analysis that was carried out?

When the downstream user wants to reproduce an experiment. Available memory and cores affect the non-determinism in algorithms. I've seen read aligners produce different alignments based on the number of cores they were run on. But I shouldn't even need to defend this. Why are we collecting this type of information if it is as useless as you suggest.

In a nutshell, I want the full manifest to be as comprehensive as technically possible. I proposed a compromise at the end of my [previous](https://github.com/DataBiosphere/azul/issues/1368#issuecomment-539868124) comment. Beyond that I'm more open to **adding** fields to the compact manifest, than to removing them from the full manifest. Also note that I am open to your idea of giving the user a chance to select which columns the manifest should contain. I think that's a great solution AND it gives us a chance to link to the portal documentation on each of the fields.

**Comment by mshadbolt** [comment]
Ok I think we have to agree to disagree. There is no way I think a user should have to read documentation to understand which column uniquely identifies a biomaterial or protocol, it should be self evident from the column title. The general attitude that "if the user can't understand our system then it is their fault" is a bit counter-productive I think, because the current system is not easy to understand, even for people inside the project. 

I have zero idea why those analysis fields are collected or shown to users, it is metadata that the analysis pipelines generate, not that the metadata team collects. My guess would be that is metadata the pipelines team would find it useful when they are looking back at their analyses rather than being collected with the aim of showing end users. Perhaps the fields aren't useless but 99% of metadata consumers would have no interest in it. 

I agree that the best compromise is giving the ability to the user to choose their metadata fields from all fields available as you suggest. I am ok with closing this issue and making a new issue to that aim.

**Comment by hannes-ucsc** [comment]
> There is no way I think a user should have to read documentation to understand which column uniquely identifies a biomaterial or protocol, it should be self evident from the column title.

This is conflating two issues. This issue is about removing columns, not changing column names.

> The general attitude that "if the user can't understand our system then it is their fault" is a bit counter-productive I think, 

That attitude is indeed counter-productive and, I assure you, is not mine. I don't think anything I said should be interpreted as if it were. All I am asking is that consumers of a data format encoding a complex and rich metadata model consult the documentation for said data format. 

Let me give a bit of background on the history of the full manifest. Initially, there was only what we now call the _compact_ manifest. It had a limited selection of columns,  a selection that mirrored what the Data Browser displayed. To get all metadata one needed to download the bundles and open the raw JSON metadata files. About a year ago, there was very strong sentiment that we would need to offer users a way to obtain all metadata in TSV format. When I offered to add more columns to the compact manifest, I faced considerable opposition. People insisted that someone would need to generate a TSV containing all (!) metadata for every project. Even though this feature didn't exactly fall into the Data Browser charter, I volunteered my team to take this on, primarily because our system architecture was a good fit for its implementation. 

Given this background, it should be unsurprising that I am pushing back a little against the removal of a large number of columns.

> I have zero idea why those analysis fields are collected or shown to users, 

I gave an example above. It sounds like you are mainly seeing this issue through the lense of a consumer of primary data. `analysis_process` is the main metadata entity in analysis bundles. That is crucial information to anyone wanting to understand how the analysis was done. 

> Perhaps the fields aren't useless but 99% of metadata consumers would have no interest in it.

I find this claim to be a bit speculative. What evidence is leading to this conclusion?

As soon as I remove the columns someone else is going to come along, asking why I am hiding this information. Your best shot at convincing me to remove them is to provide evidence and find other users that back you up. Until then, I am going to err on the side of including data.

I am reopening this ticket because I am still open to removing the schema version columns.

**Comment by theathorn** [comment]
Obsolete DCP/1 issue.
